HIP/hip-991.md

```diff
@@ -203,35 +204,43 @@ message ConsensusUpdateTopicTransactionBody {
   /*
    * The custom fee to be assessed during a message submission to this topic. Null if the fees should not be updated.
    */
-  ConsensusCustomFeeList custom_fees = 12;
+  FixedCustomFeeList custom_fees = 12;
 
 }
 
 message FeeExemptKeyList {
   repeated Key keys = 1;
 }
 
-message ConsensusCustomFeeList {
-  repeated ConsensusCustomFee fees = 1;
+message FixedCustomFeeList {
+  repeated FixedCustomFee fees = 1;
 }
 ```
 
-#### ConsensusSubmitMessageTransactionBody
+#### TransactionBody
 
-The `ConsensusSubmitMessageTransactionBody` message is updated to include the optional `max_custom_fees` property for specifying the maximum fee that the user is willing to pay for the message.
+The `TransactionBody` message is updated to include the optional `max_custom_fees` property for specifying the maximum custom fee that the user is willing to pay for the transaction.
 
 ```protobuf
-message ConsensusSubmitMessageTransactionBody {
+message TransactionBody {
   [..]
   /**
-    * The maximum custom fee that the user is willing to pay for the message. This field will be ignored if `accept_all_custom_fees` is set to `true`.
+    * The maximum custom fee that the user is willing to pay for the message. If left empty, the user is willing to pay any custom fee.
+    * If used with a transaction type that does not support custom fee limits, the transaction will fail.
     */
-  repeated FixedFee max_custom_fees = 4;
-  
+  repeated CustomFeeLimit max_custom_fees = 1001;
+}
+
+message CustomFeeLimit {
   /**
-   * If set to true, the transaction will accept all custom fees from the topic id
+   * The account of the fee payer
    */
-  bool accept_all_custom_fees = 5;
+  AccountID account_id = 1;
```

Comment:
is this supposed to be the fee collector? and if so, for consistency as named in proto message `CustomFee`, should we also name it `fee_collector_account_id`?

Reply:
+1

Reply:
No, this is not for the fee collector. This field defines the account for which the custom fees are applied. This could be the payer of the transaction or any account that appears in the `TransferList` (in the case of `CryptoTransfer`).

I'd be happy to use a more specific name, but I cannot think of anything. Using `payer` in any form would make it too easy to be confused with the payer of the transaction.